File: android/app/src/main/kotlin/com/goalmaker/app/application/sync/RemoteRejectedException.kt
package com.goalmaker.app.application.sync

/**
 * The server refused one row or call (a policy or constraint). The rest of the sync goes on.
 * [code] is the database's SQLSTATE when the server named one, so callers can tell refusals apart.
 */
class RemoteRejectedException(message: String, val code: String? = null) : Exception(message)

File: windows/src/GoalMaker.Core/Sync/RemoteRejectedException.cs
namespace GoalMaker.Core.Sync;

/// <summary>
/// The server refused a row or call for good (row security, a check); retrying the same thing won't help.
/// <see cref="Code"/> is the database's SQLSTATE when the server named one, so callers can tell refusals apart.
/// </summary>
public sealed class RemoteRejectedException(string message, string? code = null) : Exception(message)
{
    public string? Code { get; } = code;
}
